iofog-agent-daemon/src/main/java/org/eclipse/iofog/utils/logging/LogFormatter.java

```diff
@@ -28,12 +37,27 @@
  */
 public class LogFormatter extends Formatter {
 	public String format(LogRecord record) {
+		IOFogNetworkInterfaceManager fogNetworkInterfaceManager = IOFogNetworkInterfaceManager.getInstance();
 		final DateFormat df = new SimpleDateFormat("MM/dd/yyyy hh:mm:ss.SSS");
-		StringBuilder builder = new StringBuilder();
-		builder.append("[").append(df.format(System.currentTimeMillis())).append("] ");
-		builder.append("[").append(record.getLevel().toString()).append("] ");
-		builder.append(formatMessage(record)).append('\n');
-		return builder.toString();
+		JsonBuilderFactory factory = Json.createBuilderFactory(null);
+		JsonObjectBuilder jsonObjectBuilder = factory.createObjectBuilder();
+		jsonObjectBuilder.add("timestamp", df.format(System.currentTimeMillis()));
+		jsonObjectBuilder.add("level", record.getLevel().toString());
+		jsonObjectBuilder.add("agent_id", Configuration.getIofogUuid());
+		jsonObjectBuilder.add("pid", fogNetworkInterfaceManager.getPid() != 0 ? fogNetworkInterfaceManager.getPid() : fogNetworkInterfaceManager.getFogPid());
+		jsonObjectBuilder.add("hostname", fogNetworkInterfaceManager.getHostName() != null ? fogNetworkInterfaceManager.getHostName(): IOFogNetworkInterface.getHostName());
+		String[] data = formatMessage(record).split(",");
```

Comment:
I am not sure if this is the right way do extract message/stacktrace/... 
What if any of those contain `,`?
Looking at `LogRecord` class, seems like it already has `threadID`, `thrown`, `loggerName`, `millis`, ... that we could probably use

Reply:
Updated with above comments